src/main/kotlin/com/philips/hsdp/apis/iam/oauth2/IamOAuth2.kt

```diff
@@ -293,7 +297,7 @@ class IamOAuth2(
                 .post(postBody)
                 .build()
 
-            httpClient.performRequest(request, continuation, logger) { _, _ ->
+            httpClient.performRequest(request, continuation, logger) {
```

Comment:
The Android application uses internal storage to "cache" the token. The idea is that when terminating the app and reopening it again a short time later, that the app does not require you to log in again but uses the token from previous login.
By returning the revoked token (fields null and timestamp reset), it can be stored as such in the app, and upon restart/reopening the app it can read it from storage and invoke the isValid method on it. If no longer valid, then it will have to either refresh the token, or log in again.